core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -842,18 +844,42 @@ private static SortOrder updateSortOrderSchema(Schema schema, SortOrder sortOrde
     return builder.build();
   }
 
-  private static PartitionSpec freshSpec(int specId, Schema schema, PartitionSpec partitionSpec) {
+  private static PartitionSpec freshSpec(int specId, Schema schema, PartitionSpec partitionSpec, int formatVersion,
+                                         List<PartitionSpec> specs, AtomicInteger lastPartitionId) {
```

Comment:
I noted above that I don't think that we should change the behavior of `freshSpec` for `updatePartitionSpec`. Instead, what about creating a `reassignPartitionIds` method that gets called after `freshSpec` in `buildReplacement`? That would be similar to how schemas are handled, where the existing schema is passed to seed the IDs for the new table schema.

Reply:
Thanks for the comments. Updated it accordingly.